src/noLifeCycleCallRule.ts

```diff
@@ -49,19 +50,35 @@ export class ExpressionCallMetadataWalker extends NgWalker {
   }
 
   private validateCallExpression(node: ts.CallExpression): void {
-    const name = (node.expression as any).name;
+    const name = ts.isPropertyAccessExpression(node.expression) ? node.expression.name : undefined;
+    const expression = ts.isPropertyAccessExpression(node.expression) ? node.expression.expression : undefined;
 
```

Comment:
nit: extra new line.

Reply:
I intentionally separated these variable declarations for readability. Source node info vs. rule logic checks.